kernel/nativelib.ml

```diff
@@ -44,7 +71,8 @@ let rt1 = ref (dummy_value ())
 let rt2 = ref (dummy_value ())
 
 let get_ml_filename () =
-  let filename = Filename.temp_file "Coq_native" source_ext in
+  let temp_dir = Lazy.force my_temp_dir in
+  let filename = Filename.temp_file ~temp_dir "Coq_native" source_ext in
```

Comment:
If `Filename.temp_file` doesn't create non-existent directories, maybe we should make sure the directory exists here and create it if it doesn't, before returning the file name?

Reply:
`temp_file` does not create anything, it only returns the name, right? So I don't see any problem here.

Reply:
temp_file does create the file.

Reply:
So do you think it is worth first creating the directory, as @JasonGross suggests?

Reply:
>So do you think it is worth first creating the directory, as @JasonGross suggests?

I'm missing some understanding as indicated by my last comment (in the main thread on this PR, not this code comment subthread)